enterprise/causal-clustering/src/main/java/org/neo4j/causalclustering/readreplica/ReadReplicaStartupProcess.java

```diff
@@ -41,18 +42,23 @@
 
 class ReadReplicaStartupProcess implements Lifecycle
 {
+    private static final int MAX_ATTEMPTS = 5;
```

Comment:
Since this MAX_ATTEMPTS was introduced in this PR, I think it'd be nice to compute MAX_ATTEMPTS from some timeout set in config.